datamodel/changelogs/0001/02_sys.sql

```diff
@@ -21,3 +21,120 @@ BEGIN
  RETURN NEW;
 END;
 $$ LANGUAGE PLPGSQL;
+
+
+-----------------------------------------------------------------------
+-- Enable or disable Symbology Triggers
+-- To temporarily disable these cache refreshes for batch jobs like migrations
+-----------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION tww_sys.alter_symbology_triggers(action_name text) RETURNS VOID AS
+$DO$
+DECLARE
+	schdf text;
+	tbldf text;
+	trig text;
+BEGIN
+IF NOT (action_name ILIKE ANY(ARRAY['ENABLE','DISABLE'])) THEN
+	RAISE NOTICE '% not a valid input',action_name;
+	RETURN;
+ELSE
+	FOR schdf,tbldf, trig IN
+		SELECT
+		c.relnamespace ::regnamespace::text,
+		c.relname,
+		t.tgname
+		FROM   pg_trigger t
+		INNER JOIN pg_class c on t.tgrelid=c.oid
+		INNER JOIN pg_proc p on t.tgfoid=p.oid
+		WHERE p.proname  LIKE 'symbology_%'
+		and p.pronamespace::regnamespace::text LIKE 'tww_ap_'
+		LOOP
+			EXECUTE FORMAT('ALTER TABLE %I.%I %s TRIGGER %I',schdf,tbldf,upper(action_name),trig);
+		END LOOP;
+		RETURN;
+END IF;
+END;
+$DO$
+LANGUAGE plpgsql SECURITY DEFINER;
+
+
+-----------------------------------------------------------------------
+-- Check if Symbology Triggers are enabled
+-----------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION tww_sys.check_symbology_triggers_enabled() RETURNS BOOL AS
+$DO$
+DECLARE _disabled_count numeric;
+BEGIN
+  SELECT count(*) into _disabled_count
+		FROM   pg_trigger t
+		INNER JOIN pg_class c on t.tgrelid=c.oid
+		INNER JOIN pg_proc p on t.tgfoid=p.oid
+		WHERE p.proname  LIKE 'symbology_%'
+		and p.pronamespace::regnamespace::text LIKE 'tww_ap_'
+		AND t.tgenabled = 'D';
+  IF _disabled_count=0 THEN
+    return true;
+  ELSE
+    return false;
+  END IF;
+END;
+$DO$ LANGUAGE plpgsql SECURITY DEFINER;
+
+
```

Comment:
I'm no great fan of auto_update term, can we define some better naming for this?

We talk about triggers that set a new modification date and user to any change in the database right?

I would talk about `system_trigger` or `modification_trigger`. The fact that it is automated is in my understanding a bit implicit since it is a trigger.